src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -230,46 +231,52 @@ public Query termQuery(Object value, QueryShardContext context) {
 
     @Override
     public void parse(ParseContext context) throws IOException {
-        if (!KNNSettings.isKNNPluginEnabled()) {
-            throw new IllegalStateException("KNN plugin is disabled. To enable " +
-                                                    "update knn.plugin.enabled setting to true");
-        }
+        KNNCounter.GRAPH_INDEX_REQUESTS.increment();
```

Comment:
This might add a little confusion for bulk requests, which could index multiple vectors but still part of same request. If the intention is to count the number of graph requests , probably we could count at ```KNN80DocValuesConsumer```.

Reply:
Yes, the intention of this metric is to count the total number of requests to index graphs. Will move to `KNN80DocValuesConsumer`